setup.py

```diff
@@ -77,12 +76,12 @@ def run(self):
 
     def run_tests(self):
         import tempfile
-        fd, filename = tempfile.mkstemp(prefix='temprunner', text=True)
-        scriptfile = open(filename, 'w')
         script = CUSTOM_TEST_TEMPLATE % (
             sys.path, os.path.abspath(os.curdir), os.path.abspath('src'))
-        scriptfile.write(script)
-        scriptfile.close()
+        _fd, filename = tempfile.mkstemp(prefix='temprunner', text=True)
+        with open(filename, 'w') as scriptfile:
+            scriptfile.write(script)
+            scriptfile.close()
```

Comment:
You shouldn't need an explicit `close()` here.

Reply:
D'oh!